### PR TITLE
trellis_m4: ADXL343 accelerometer support (and examples)

### PR DESCRIPTION
**Pull request by tarcieri**
Uses my recently released `adxl343` crate:

https://github.com/NeoBirth/ADXL343.rs

Includes the following example to visualize the accelerometer output using the NeoPixels:

ezgif-1-9dd510d18d2d

### CODE REVIEW

File: boards/trellis_m4/src/lib.rs [hunk 1]
```diff
@@ -12,19 +15,24 @@ pub extern crate keypad;
 
 pub mod pins;
 
+/// Adxl343 (preconfigured to use SERCOM2)
+#[cfg(feature = "adxl343")]
+type Adxl343 = adxl343::Adxl343<hal::sercom::I2CMaster2>;
```
Comment (tarcieri) [inline boards/trellis_m4/src/lib.rs]: It might be a bit draconian to require people use SERCOM2...

File: boards/trellis_m4/src/lib.rs [hunk 2]
```diff
@@ -12,19 +15,24 @@ pub extern crate keypad;
...
 
 use hal::prelude::*;
-use hal::clock::GenericClockController;
-use hal::sercom::{I2CMaster4, PadPin};
-use hal::time::Hertz;
 #[cfg(feature = "keypad-unproven")]
 use hal::gpio::{OpenDrain, Output, PullUp};
 
+/// Number of Neopixels on the device
+pub const NEOPIXEL_COUNT: usize = 32;
```
Comment (tarcieri) [inline boards/trellis_m4/src/lib.rs]: Figured this could use its own constant

File: boards/trellis_m4/src/lib.rs [hunk 3]
```diff
@@ -212,25 +220,3 @@ impl Keypad {
...
-    let gclk0 = clocks.gclk0();
-    I2CMaster4::new(
-        &clocks.sercom4_core(&gclk0).unwrap(),
-        bus_speed.into(),
-        sercom4,
-        mclk,
-        sda.into_pad(port),
-        scl.into_pad(port),
-    )
-}
```
Comment (tarcieri) [inline boards/trellis_m4/src/lib.rs]: With multiple I2C devices I thought it'd make sense to move this into the corresponding pins type. It certainly helps shave off a few arguments!

File: boards/trellis_m4/src/pins.rs [hunk 4]
```diff
@@ -36,6 +44,41 @@ pub struct Accelerometer {
     pub scl: Pa13<Input<Floating>>,
 }
 
+impl Accelerometer {
```
Comment (tarcieri) [inline boards/trellis_m4/src/pins.rs]: Potentially we could add some generics with default values to this type to allow for use with SERCOMs other than 2...

File: boards/trellis_m4/src/lib.rs (hunk 3 above)
Comment (sajattack) [reply boards/trellis_m4/src/lib.rs]: Could you please re-export the new i2c_master function in it's place?

Comment (tarcieri) [reply boards/trellis_m4/src/lib.rs]: Done

File: boards/trellis_m4/src/pins.rs (hunk 4 above)
Comment (tarcieri) [reply boards/trellis_m4/src/pins.rs]: Tried that and wasn't happy with the result. SERCOM4 is the only other one compatible with the accelerometer pins, and the original `i2c_master` function is already hardcoded to that...

**Pull request merged**